examples/pic.cpp

```diff
@@ -0,0 +1,203 @@
+
+    int maxIter = std::stoi(husky::Context::get_param("maxIter"));
+    float epsilon = std::stod(husky::Context::get_param("stopThres"));
+
+    int outPerNIter = std::stoi(husky::Context::get_param("outPerNIter"));
+
+    if (maxIter < 0)
+        maxIter = 10;
+    if (epsilon < 0)
+        epsilon = 1e-5 / totDataPoints;
+
+    for (uint iter = 0; iter < maxIter; ++iter) {
+        list_execute(node_list, [&](Node& n) {
+            // here we assuemed the edge is bidirectional with same weight (a.k.a Affinity mat is symmetric)
+            for (uint i = 0; i < n.id(); ++i) {
+                nodeTonodeCh.push(n.edgeW[i] * n.val, i);
+            }
+            for (uint i = n.id() + 1; i < totDataPoints; ++i) {
+                nodeTonodeCh.push(n.edgeW[i] * n.val, i);
+            }
+        });
+
+        list_execute(node_list, {&nodeTonodeCh}, {&ac}, [&](Node& n) {
+            n.valtmp = n.edgeWsum > 0 ? nodeTonodeCh.get(n) / n.edgeWsum : 0.0;
+            normFacAgg.update(n.valtmp);
+        });
+
+        list_execute(node_list, {}, {&ac}, [&](Node& n) {
+            n.setVal(n.valtmp / normFacAgg.get_value());
+            maxFloatAgg.update(n.getAcc());
+            if (outPerNIter > 0 && iter % outPerNIter == 0) {
+                std::ostringstream oss;
+                oss.precision(17);
+                oss << n.id() << " " << n.val << "\n";
+                husky::io::HDFS::Write("master", "9000", oss.str(),
+                                       husky::Context::get_param("outDir") + "/iter" + std::to_string(iter),
+                                       husky::Context::get_global_tid());
+            }
+        });
+
+        if (husky::Context::get_global_tid() == 0) {
+            husky::base::log_msg("Iter " + std::to_string(iter) + " " + "MaxAcc " +
+                                 std::to_string(maxFloatAgg.get_value()) + " " + "log10(MaxAcc) " +
+                                 std::to_string(log10(maxFloatAgg.get_value())));
+        }
+
+        if (maxFloatAgg.get_value() < epsilon)
+            break;
+        maxFloatAgg.to_reset_each_iter();
+        normFacAgg.to_reset_each_iter();
+    }  // end iter loop
+
+    // 4. output
+    list_execute(node_list, [&](Node& n) {
+        std::ostringstream oss;
+        oss.precision(17);
+        oss << n.id() << " " << n.val << "\n";
+        husky::io::HDFS::Write("master", "9000", oss.str(), husky::Context::get_param("outDir") + "/fin",
+                               husky::Context::get_global_tid());
+    });
+}
+
+int main(int argc, char** argv) {
+    std::vector<std::string> args;
+    args.push_back("hdfs_namenode");
+    args.push_back("hdfs_namenode_port");
+    args.push_back("input");
+    args.push_back("outDir");
+    args.push_back("maxIter");      // <=0 implies 10
+    args.push_back("stopThres");    // <=0 implies auto
+    args.push_back("outPerNIter");  // <=0 implies turn off
```

Comment:
Please indicate the meaning of these parameters, since the users have to pass them to your code.